packages/backend/src/graphql/resolvers/tenant.test.ts

```diff
@@ -0,0 +1,513 @@
+        ${true}    | ${'operator'}
+        ${false}   | ${'tenant'}
+      `(
+        'can update a tenant as $description',
+        async ({ isOperator }): Promise<void> => {
+          const client = isOperator
+            ? appContainer.apolloClient
+            : tenantedApolloClient
+          const updateInput = {
+            ...generateTenantInput(),
+            id: tenant.id
+          }
+
+          const mutation = await client
+            .mutate({
+              mutation: gql`
+                mutation UpdateTenant($input: UpdateTenantInput!) {
+                  updateTenant(input: $input) {
+                    tenant {
+                      id
+                      email
+                      apiSecret
+                      idpConsentUrl
+                      idpSecret
+                      publicName
+                    }
+                  }
+                }
+              `,
+              variables: {
+                input: updateInput
+              }
+            })
+            .then((query): TenantMutationResponse => query.data?.updateTenant)
+
+          expect(mutation.tenant).toEqual({
+            ...updateInput,
+            __typename: 'Tenant'
+          })
+        }
+      )
+      test('Cannot update other tenant as non-operator', async (): Promise<void> => {
+        const firstTenant = await createTenant(deps)
+        const secondTenant = await createTenant(deps)
+
+        const updateInput = {
+          ...generateTenantInput(),
+          id: secondTenant.id
+        }
+        const client = createTenantedApolloClient(appContainer, firstTenant.id)
+        try {
+          expect.assertions(2)
+          await client
+            .mutate({
+              mutation: gql`
+                mutation UpdateTenant($input: UpdateTenantInput!) {
+                  updateTenant(input: $input) {
+                    tenant {
+                      id
+                      email
+                      apiSecret
+                      idpConsentUrl
+                      idpSecret
+                      publicName
+                    }
+                  }
+                }
+              `,
+              variables: {
+                input: updateInput
+              }
+            })
+            .then((query): TenantMutationResponse => query.data?.updateTenant)
+        } catch (error) {
+          expect(error).toBeInstanceOf(ApolloError)
+          expect((error as ApolloError).graphQLErrors).toContainEqual(
+            expect.objectContaining({
+              message: 'tenant does not exist',
+              extensions: expect.objectContaining({
+                code: GraphQLErrorCode.NotFound
+              })
+            })
+          )
+        }
+      })
+    })
+
+    describe('Delete', (): void => {
+      test.each`
+        isOperator | description
+        ${true}    | ${'operator'}
+        ${false}   | ${'tenant'}
+      `(
+        'Can delete a tenant as $description',
```

Comment:
Does this mean a tenant can delete itself? 
If so, do you think should it be reserved only for operators?

Reply:
Making deleteTenant operator-only as per call today.